homeassistant/components/recorder/models.py

```diff
@@ -198,6 +210,33 @@ def to_native(self, validate_entity_id=True):
             return None
 
 
+class Statistics(Base):  # type: ignore
+    """Statistics."""
+
+    __table_args__ = {
+        "mysql_default_charset": "utf8mb4",
+        "mysql_collate": "utf8mb4_unicode_ci",
+    }
+    __tablename__ = TABLE_STATISTICS
+    id = Column(Integer, primary_key=True)
+    entity_id = Column(String(255))
+    period = Column(Enum(*STATISTIC_PERIODS))
+    start = Column(DATETIME_TYPE, index=True)
+    mean = Column(Float())
+    min = Column(Float())
+    max = Column(Float())
```

Comment:
There are two ways to structure this data. We can have each aggregation type be their own column or we have an `aggregation_type` and `value` column, with type being an enum of `mean`, `min`, `max`. The latter would make it easier in the future to add more types. `cost` is one that comes to mind.

Reply:
If we do just:
`| id | statistic_id | period | start | statistic_type | value |`
there will be a very big overhead.

Modified in 407dc9d3905887511ac3fd8c6448ba6e08f9f3d4 so the `statistics` table has these columns:
`| id | statistic_id | period | start | statistic_type | value_1 | value_2 | value_3 |`

`statistic_type` controls the mapping of statistics to the value columns

We can add additional value columns if needed.